### PR TITLE
Change typing of ItemSeparatorComponent to React.ComponentType for SectionListProperties

### PR DESCRIPTION
**Pull request by thorbenandresen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

Typing have already been fixed for the FlatListProperties (see links below) so this PR is to fix the SectionListProperties
 * https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18924
 * https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18823

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -3728,7 +3728,7 @@ export interface SectionListProperties<ItemT> extends ScrollViewProperties {
     /**
      * Rendered in between adjacent Items within each section.
      */
-    ItemSeparatorComponent?: React.ComponentClass<any> | (() => React.ReactElement<any>) | null
+    ItemSeparatorComponent?: React.ComponentType<any> | (() => React.ReactElement<any>) | null
```
Comment (alloy) [inline types/react-native/index.d.ts]: Please remove the second type in the union, which is a part of `ComponentType`.

**Review by alloy — CHANGES_REQUESTED** [review]

File: types/react-native/index.d.ts (hunk 1 above)
Comment (Thorbenandresen) [reply types/react-native/index.d.ts]: done, thanks!

**Comment by timwangdev** [comment]
👍

**Comment by alloy** [comment]
Thanks!

**Review by alloy — APPROVED** [review]

**Comment by alloy** [comment]
Aww, @typescript-bot. Getting overworked, ey?

**Pull request merged**